app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt

```diff
@@ -101,6 +110,45 @@ class MessageActivity : AppCompatActivity() {
             }
         }
 
+        save.setOnClickListener {
+            try {
+                val inputManager: InputMethodManager = this.getSystemService(Context.INPUT_METHOD_SERVICE)
+                        as InputMethodManager
+                inputManager.hideSoftInputFromWindow(content.windowToken, InputMethodManager.SHOW_FORCED)
+
+                val sdf = SimpleDateFormat("dd-MMM-yyyy-hh_mm_aaa")
+                val fileName = "badge-magic-" + sdf.format(Calendar.getInstance().time).toString()
+                val os = FileOutputStream(fileName, true)
+                var text = "null"
+                var drawableID = "null"
+                var flashSelected = false
+                var marqueeSelected = false
+
+                if (content.text.isNotBlank())
+                    text = content.text.toString()
+                else if (drawableRecyclerAdapter.selectedPosition >= 0)
+                    drawableID = drawableRecyclerAdapter.selectedPosition.toString()
+
+                val selectedSpeed = speed.selectedItem.toString()
+                val selectedMode = mode.selectedItem.toString()
+
+                when {
+                    flash.isChecked -> flashSelected = true
+                    marquee.isChecked -> marqueeSelected = true
```

Comment:
Why are you using this even? when is not necessary

Reply:
Because after that only it will be passed in the Moshi Converter

Reply:
you can directly use `flash.isChecked` anywhere you want. No need to check if it is true or not, Because assigning false initially and then checking if the flash is checked or not is redundant. `flash.isChecked` already has the boolean you need

Reply:
Yeah right... I'll change that after my lectures ! Thanks !